tests/integration/simulacron/test_connection.py

```diff
@@ -178,6 +178,79 @@ def test_callbacks_and_pool_when_oto(self):
         errback.assert_called_once()
         callback.assert_not_called()
 
+    def test_heartbeat_defunct_deadlock(self):
```

Comment:
I can't reproduce this issue if the python driver is not build with cython. Add the `@cythontest` decorator here. --> from tests.unit.cython.utils import cythontest